app.rb

```diff
@@ -1,124 +1,110 @@
+  choice = gets.chomp.to_i
+  print 'Enter age: '
+  age = gets.chomp.to_i
+  print 'Enter name: '
+  name = gets.chomp
+  print 'Does the person have parent permission? (y/n): '
+  parent_permission = gets.chomp.downcase == 'y'
+
+  case choice
+  when 1
+    person = Student.new(people.length + 1, age, name, parent_permission)
+  when 2
+    create_teacher(people, age, name, parent_permission)
+    return
+  else
+    puts 'Invalid choice'
+    return
   end
 
-  def create_rental
-    puts 'Select a book from the following list by number'
-    @books.each_with_index do |book, index|
-      puts "#{index}) Title: #{book.title}, Author: #{book.author}"
-    end
+  people << person
+  puts "Created person #{person.name}"
+end
 
-    book_id = gets.chomp.to_i
+def create_teacher(people, age, name, parent_permission)
+  print 'Enter specialization: '
+  specialization = gets.chomp
+  person = Teacher.new(people.length + 1, age, name, parent_permission, specialization)
+  people << person
+  puts "Created teacher #{person.name}"
+end
 
-    puts 'Select a person from the following list by number (not ID)'
-    @people.each_with_index do |person, index|
-      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
-    end
+# Create a book
+def create_book(books)
+  print 'Enter title: '
+  title = gets.chomp
+  print 'Enter author: '
+  author = gets.chomp
+  book = Book.new(title, author)
+  books << book
+  puts "Created book #{book.title} by #{book.author}"
+end
 
-    person_id = gets.chomp.to_i
+# Create a rental
+def create_rental(people, books, rentals)
+  puts 'Select a book from the following list by number:'
+  books.each { |book| puts "[#{book.id}] #{book.title} by #{book.author}" }
+  book_id = gets.chomp.to_i
+  book = books.find { |b| b.id == book_id }
 
-    print 'Date: '
-    date = gets.chomp
+  return puts 'Book not found' unless book
 
-    @rentals << Rental.new(date, @people[person_id], @books[book_id])
-    puts 'Rental created successfully'
-  end
+  puts 'Select a person from the following list by number:'
+  people.each { |person| puts "[#{person.id}] #{person.name} (#{person.class})" }
+  person_id = gets.chomp.to_i
+  person = people.find { |p| p.id == person_id }
 
-  def list_rentals
-    print 'ID of person: '
-    id = gets.chomp.to_i
+  return puts 'Person not found' unless person
 
-    rentals = @rentals.filter { |rental| rental.person.id == id }
+  print 'Enter the rental date (YYYY-MM-DD): '
+  rental_date = gets.chomp
+  rental = Rental.new(person, book, rental_date)
+  rentals << rental
+  puts "Created rental of '#{book.title}' by #{book.author} to #{person.name} on #{rental_date}"
+end
 
-    puts 'Rentals:'
-    rentals.each do |rental|
-      puts "Date: #{rental.date}, Book: '#{rental.book.title}' by #{rental.book.author}"
+def list_rentals(people, rentals)
+  puts 'Enter ID of person:'
+  person_id = gets.chomp.to_i
+  person = people.find { |p| p.id == person_id }
```

Comment:
- [ ] as the previous CR mentioned any code in relation to input and input validation should be placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input. this includes all the `gets.chomp` parts of the code.